extensions/android-paging3/src/main/java/com/squareup/sqldelight/android/paging3/OffsetQueryPagingSource.kt

```diff
@@ -33,11 +33,13 @@ internal class OffsetQueryPagingSource<RowType : Any>(
   override suspend fun load(
     params: LoadParams<Long>
   ): LoadResult<Long, RowType> = withContext(dispatcher) {
+    val key = when {
+      params is LoadParams.Refresh -> (params.key ?: 0L) / params.loadSize * params.loadSize
+      else -> params.key ?: 0L
+    }
     try {
-      val key = params.key ?: 0L
       transacter.transactionWithResult {
         val count = countQuery.executeAsOne()
-        if (count != 0L && key >= count) throw IndexOutOfBoundsException()
```

Comment:
I had to remove this (https://github.com/cashapp/sqldelight/pull/2219/files#r825447672), consider the following scenario:

You have 3 items from your database. All are currently shown because some `column = true` when changing that particular column for instance for the last entry, the item should not be shown any longer since `column = false`. This used to work fine in paging2, but now it crashes here. 

Removing it just works fine. The item won't be shown and my RecyclerView only has 2 items.